config/v1/types_infrastructure.go

```diff
@@ -354,6 +354,11 @@ type AWSPlatformSpec struct {
 	// There must be only one ServiceEndpoint for a service.
 	// +optional
 	ServiceEndpoints []AWSServiceEndpoint `json:"serviceEndpoints,omitempty"`
+
+	// LBType allows user to set a load balancer type.
+	// Allowed values are nlb or elb
+	// +optional
+	LBType string `json:"lbType,omitempty"`
```

Comment:
You should create an enum rather than a plain string, ie a typed string, which has some exported constants with the possible values,

Given you're adding some load balancer configuration, is there any change the user may want to be able to configure any of the load balancer options, ELB or NLB at some point in the future?

I think this probably wants to end up as a discriminated union with configuration for both types of load balancer